django_deprecate_fields/deprecate_field.py

```diff
@@ -16,18 +16,18 @@ def deprecate_field(field_instance, return_instead=None):
     :param return_instead: A value or function that
     the field will pretend to have
     """
-    if set(sys.argv) & {"makemigrations", "migrate"}:
-        if type(field_instance) == BooleanField:
-            # A BooleanField does not allow null=True, so we need to cast
-            # this to a NullBooleanField
-            return NullBooleanField(
-                help_text=field_instance.help_text,
-                default=field_instance.default
-            )
-        else:
-            field_instance.null = True
-            return field_instance
-    else:
-        if callable(return_instead):
-            return return_instead()
-        return return_instead
+    if not set(sys.argv) is {"makemigrations", "migrate"}:
```

Comment:
Adding this `is` no longer returns the opposite of `if set(sys.argv) & {"makemigrations", "migrate"}:` - is this correct?

Reply:
oh oh. I think we might need something like 
``
if not 'makemigrations' in list(sys.argv) or not 'migrate' in list(sys.argv):
```

Reply:
btw, `B & {"makemigrations", "migrate"}`  will always return true.

Reply:
Wouldn't it work to just do `if not set(sys.argv) & {"makemigrations", "migrate"}`?

Reply:
Im confused, because this seems correct when I run it through the shell:
```python
In [2]: not set(['migrate']) & {"makemigrations", "migrate"}
Out[2]: False

In [3]: not set(['notinsysv']) & {"makemigrations", "migrate"}
Out[3]: True
```

I don't think i've used the `&` operator much so I can't claim to be expert on it :man_shrugging:

Reply:
I also think making the other arg a set `not set(['notinsysv']) & set(["makemigrations", "migrate"])` is better here

Reply:
@justinline : okey. Let me try to explain this here. 
![image](https://user-images.githubusercontent.com/4214481/46949650-714d0900-d082-11e8-87b2-b44e76b05bab.png)
`sys.argv` -> List 

Now, you need to check if `migrate` or `makemigrations` are in this `sys.argv`. 

So you take `migrate` or `makemigrations` and check if they are in sys.argv. 

Now, the code: 
```
if set(sys.argv) & {"makemigrations", "migrate"}:
``` 
is similar to 
```
if {'/usr/bin/ipython'} &  {"makemigrations", "migrate"}: #set(list) -> dict()
```
Now the RHS of your condition is always **True**. This would make it look like: 
```
if {'/usr/bin/ipython'} &  True
```
which would make this work for every management command from the command line. Say for `python manage.py createsuperuser`. 

I hope this helps.

Reply:
My apologies. Looks like `set()` does some magic with `&` operation. @justinline is :heavy_check_mark: